### DISCUSSION TITLE
All 276 Styles in Fooocus (Google table)

### DISCUSSION THREAD

**Discussion by Irmagon** [lllyasviel/Fooocus#2082] [Show and tell]
Hi all.
The previous tables [1](https://github.com/lllyasviel/Fooocus/discussions/143), [2](https://github.com/lllyasviel/Fooocus/issues/140) are now somewhat outdated.
Made a new spreadsheet today, with all 276 styles available.
4 prompts, juggernautXL_v8 and SSD-1B_LCM models, RAW positive/negative style additional prompts.

https://docs.google.com/spreadsheets/d/10UXU59fBA5wZrcEEvgv1MTf1MSNTCB5PT4fNxsAst8E

![image](https://github.com/lllyasviel/Fooocus/assets/36661892/5ef1a89d-3c61-4fb7-9369-369864069a49)

![image](https://github.com/lllyasviel/Fooocus/assets/36661892/c14c7985-78f5-4d22-8ab2-6add939d8b95)

**Comment by OceanXL** [comment]
Lot of time and effort! Thanks for sharing, just looking at these gives inspiration.

**Reply by timothy68** [reply]
Cool. Thank you for the summary and import Image sample. I like the document here.

**Comment by poor7** [comment]
1. Why Fooocus Enhance without promts?
2. What seed and size did you use?
3. Some very strange pictures that somehow barely reflect the essence of the style. For example:
Seed: 1234, Style: MRE Bad Dream, Promt: bedroom with big window
![image (3)](https://github.com/lllyasviel/Fooocus/assets/154618384/477ba21e-48d4-4b3b-9428-c5d839cf6b3f)
Compare with yours
![snap1](https://github.com/lllyasviel/Fooocus/assets/154618384/4717616d-c2b2-48e4-a4be-579495101f5a)

**Reply by Irmagon** [reply]
Hi
seed: 3
Model: SSD-1B
A picture difference - 1B is very small model, so very fast for generation
In past i regenerate styles with juggernautXL_v8 and add additional sheet in table

**Reply by poor7** [reply]
> A picture difference - 1B is very small model, so very fast for generation.

I use this model https://huggingface.co/Lykon/dreamshaper-xl-v2-turbo/resolve/main/DreamShaperXL_Turbo_v2.safetensors
In Colab it generates 1920x1080 images in 35-40 seconds (or 1024x1024 in 15-20 sec) in 8 steps with CFG:2 and Guidance Scale:2, Sampler:dpmpp_sde_gpu, the rest are all default settings.
Examples of images, Styles: MRE Dark Dream, Prompts: same
![image (4)](https://github.com/lllyasviel/Fooocus/assets/154618384/e5c8413c-be40-49db-bd56-1fe4188939b6)
![image (5)](https://github.com/lllyasviel/Fooocus/assets/154618384/428fb62e-086a-4f4b-929c-7d703a119a79)
![image (3)](https://github.com/lllyasviel/Fooocus/assets/154618384/e46c5f14-d818-4268-b52f-04410b88354e)
![image (1)](https://github.com/lllyasviel/Fooocus/assets/154618384/ffc7f22a-85f5-4d37-b885-36a20e0b6f1f)
What's the point of using a model SSD-1B?

**Reply by mashb1t** [reply]
SSD-1B is faster than speed performance setting, but similar to turbo.

**Reply by poor7** [reply]
> but _similar_ to turbo.

I wouldn't say it's "similar". He does not understand prompts at all, both positive and negative. Look at the pictures of the turbo and this “analog”.
![snap](https://github.com/lllyasviel/Fooocus/assets/154618384/e4cc0256-a41e-4f67-bad2-3678ad263c22)

**Reply by mashb1t** [reply]
SSD-1B uses less storage and in general, also less parameters. I was mainly referring to speed.

**Comment by Irmagon** [comment]
> 1. Why Fooocus Enhance without promts?
Fixed
Fooocus Enhance only add negative in final prompt.

**Comment by toutjavascript** [comment]
Hello

I am working on a python app with web UI to manage models and styles
Here an exemple of what it looks like
You can prepare as many presets as you like and launch the generation with one click
It renders like that
![Capture d’écran 2024-02-09 à 12 51 59](https://github.com/lllyasviel/Fooocus/assets/30899600/46e067b4-af97-4f91-854a-c44b7f3cc583)

And a grid view on a preset
![Capture d’écran 2024-02-09 à 12 59 31](https://github.com/lllyasviel/Fooocus/assets/30899600/89182483-7724-42c0-8888-7cca8e777417)

And the same with models 
![Capture d’écran 2024-02-09 à 12 57 02](https://github.com/lllyasviel/Fooocus/assets/30899600/e8a14e40-3425-4f2a-bd3b-dcac9f688aa4)

It works with the Fooocus gradio API.

**Reply by Irmagon** [reply]
Can you share the link to this script?
Everyone would be very grateful to you )))

**Reply by toutjavascript** [reply]
I will do on a public github 
But script is not ready yet
I hope to be ready next week

**Reply by bsdesigner** [reply]
Super!

**Reply by toutjavascript** [reply]
Hi guys,
The repo is freshly opened
https://github.com/toutjavascript/FoooXus-Fooocus-Extender

Please, tell me if it works : it is my first shared python project. So I am not sure about install process

**Reply by DmitriyL11** [reply]
how do i set it up if i use google colab focus?

**Comment by Irmagon** [comment]
> Hi seed: 3 Model: SSD-1B A picture difference - 1B is very small model, so very fast for generation 
In past i regenerate styles with juggernautXL_v8 and add additional sheet in table

Add in table output from  JuggernautXL_v8, steps:20, CFG:4

**Comment by xhoxye** [comment]
CFG:4 is more realistic for the model juggernautXL, so CFG:7 is recommended

![QQ截图20240323221659](https://github.com/lllyasviel/Fooocus/assets/129571231/a68a7150-5cc4-4325-82e2-c400d5aad152)
![QQ截图20240323221647](https://github.com/lllyasviel/Fooocus/assets/129571231/bb5333b0-14ac-424b-b7f8-511b32703ac1)

**Comment by DavidDragonsage** [comment]
@Irmagon  I have really appreciated this Styles table and recommended it to members of the [Pure Fooocus](https://www.facebook.com/groups/fooocus) Facebook group and others. I have just added a link to the table under the Style selector in [FooocusPlus](https://github.com/DavidDragonsage/FooocusPlus). There is also a credit in the Readme. These updates will become live in FooocusPlus 1.0.8. If you have any concerns, please let me know. ❤️

**Comment by Irmagon** [comment]
@DavidDragonsage 
Embedded styles from ZIT text-encoder
Many styles is specific and need to reworked!!!

https://docs.google.com/spreadsheets/d/1peQANLesZdOpAShAkC-i_jJ5rTnmeBfL

[sdxl_styles_zit.json](https://github.com/user-attachments/files/24554793/sdxl_styles_zit.json)

**Comment by DavidDragonsage** [comment]
@Irmagon  Wow, this is quite fascinating! The current Stylesheet, based on you work, uses Jugg 8 of course. While these new styles are ZIT based, they should at least be able to be integrated into what we have. And if course integrated into the FooocusPlus style selector. I shall have to think through how best to do all this.

Just a note that the Comfy upgrade for FooocusPlus is delayed until 1.1.1. It was my intention to proceed directly to the Comfy upgrade and ZIT integration in FooocusPlus 1.1.0 but Gradio 3.42.2 has caught up with me, creating bugs and frustrating limitations. So the current priority for FooocusPlus 1.1.0 is the upgrade to Gradio 6.3.0. This is proving quite challenging and time consuming, as I expected.

**Reply by Irmagon** [reply]
David 
Is it really necessary to update Gradio? As far as I can see, there are no special dependencies in the code for transitioning to the new version. It also seems that there are no deprecated dependencies in the current version.

**Reply by DavidDragonsage** [reply]
@Irmagon Well we just noticed this problem: [Pillow 11.x breaks Gradio 3.x](https://github.com/DavidDragonsage/FooocusPlus/issues/312). I have been fighting with the erratic formatting of Gradio 3.42.2 lately and this latest bug tipped it over the edge.

It is also my experience that up to 90% of bugs in FooocusPlus have been caused by Gradio. Whether or not Gradio 6.3.0 will be more stable is in question of course, but the later versions of Gradio have been tested to be more secure. Originally designed to avoid the serious security breaches of the Simple SDXL branch of Fooocus, improved security has always been one of our prime directives.

We had scheduled the Gradio upgrade to be simultaneous to the Comfy upgrade but I soon realized that introducing two disruptions at once we be unwise. Not that I expect the Comfy upgrade will be nearly as disruptive! I typically take the bitter medicine first before enjoying the pleasant dessert.

**Comment by DavidDragonsage** [comment]
@Irmagon The Gradio upgrade is ongoing and has resulted in many improvements to the codebase. However, after several months work - most of the UI has to be rewritten - I realize that it is delaying important projects like this one.

I have retrofitted several of the Gradio 5.5 improvements to the Gradio 3.43.2 version that FooocusPlus currently uses. These will be released as FooocusPlus 1.1.0, likely later this week. While it often seems that some initial unforeseen bugs accompany major code refactoring, I am optimistic that the 1.1.0 release will be the most stable one so far. This will provide a good basis for Gradio 1.1.1, which will involve the Comfy upgrade and support for Z-Image Turbo.
